bluebrain/repo-patches/packages/neuron/package.py

```diff
@@ -241,7 +242,7 @@ def cmake_enable_option(spec_requiremement):
             args.append("-DNRN_SANITIZERS=" + ",".join(self.spec.variants["sanitizers"].value))
         if "+mpi" in self.spec:
             args.append("-DNRN_ENABLE_MPI=ON")
-            if "~coreneuron" in self.spec:
+            if "~gpu" in self.spec:
```

Comment:
I'm not sure this will work as intended.

The `gpu` variant is conditional on `+coreneuron`, so I think this condition is met for `+coreneuron~gpu` but not for `~coreneuron` (because in that case there is no `gpu` variant).

I checked this with `spack install -v neuron~coreneuron` and saw
```
==> neuron: Executing phase: 'cmake'
==> [2023-08-03-11:39:38.254035] '/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/cmake-3.24.3-4blf3d/bin/cmake' '-G' 'Ninja' '-DCMAKE_INSTALL_PREFIX:STRING=/gpfs/bbp.
cscs.ch/home/olupton/spack_install/software/install_oneapi-2022.2.1-skylake/neuron-9.0.a10-hshuge' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATI
ON:BOOL=OFF' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/gpfs/bbp.cscs.ch/home/olupton/spack_install/software/install_oneapi-2022.2.1-skylake/neuron-9.0.a10-hshuge/lib;/gp
fs/bbp.cscs.ch/home/olupton/spack_install/software/install_oneapi-2022.2.1-skylake/neuron-9.0.a10-hshuge/lib64;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_applications/install_gcc-12.2.0-skylake/gette
xt-0.21.1-xztjur/lib;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/bzip2-1.0.8-e7n5ge/lib;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_applications/install_gcc-12
.2.0-skylake/libiconv-1.16-455jia/lib;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_applications/install_gcc-12.2.0-skylake/libxml2-2.10.1-sktzpi/lib;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_appl
ications/install_gcc-12.2.0-skylake/xz-5.2.7-qy7gxu/lib;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/zlib-1.2.13-7d6hpk/lib;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-2
3/stage_externals/install_gcc-12.2.0-skylake/ncurses-6.3-mslafi/lib;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/hpe-mpi-2.25.hmpt-h77oow/lib;/gpfs/bbp.cscs.ch/ssd/
apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/python-3.10.8-rvn6l5/lib;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/readline-8.1.2-kapavr/lib' '-DC
MAKE_PREFIX_PATH:STRING=/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_applications/install_gcc-12.2.0-skylake/py-sympy-1.11.1-d5vf2a;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_applications/install_
gcc-12.2.0-skylake/py-numpy-1.22.4-ly6cmf;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_applications/install_gcc-12.2.0-skylake/py-cython-0.29.32-bjojgf;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_a
pplications/install_gcc-12.2.0-skylake/py-pip-22.2.2-3wgk4c;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_applications/install_gcc-12.2.0-skylake/gettext-0.21.1-xztjur;/gpfs/bbp.cscs.ch/ssd/apps/bsd/202
3-02-23/stage_applications/install_gcc-12.2.0-skylake/libxml2-2.10.1-sktzpi;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_applications/install_gcc-12.2.0-skylake/xz-5.2.7-qy7gxu;/gpfs/bbp.cscs.ch/ssd/ap
ps/bsd/2023-02-23/stage_applications/install_gcc-12.2.0-skylake/libiconv-1.16-455jia;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/readline-8.1.2-kapavr;/gpfs/bbp.cs
cs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/python-3.10.8-rvn6l5;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/ninja-1.11.1-o5qggc;/gpfs
/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/hpe-mpi-2.25.hmpt-h77oow;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/ncurses-6.3-ms
lafi;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/zlib-1.2.13-7d6hpk;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/bzip2-1.0.
8-e7n5ge;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/flex-2.6.3-2jtkry;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/cmake-3
.24.3-4blf3d;/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/bison-3.8.2-4q6ji4' '-DNRN_ENABLE_INTERVIEWS:BOOL=FALSE' '-DNRN_ENABLE_PYTHON:BOOL=TRUE' '-DNRN_ENABLE_MEM
ACS:BOOL=TRUE' '-DNRN_ENABLE_RX3D:BOOL=TRUE' '-DNRN_ENABLE_CORENEURON:BOOL=FALSE' '-DNRN_ENABLE_TESTS:BOOL=FALSE' '-DNRN_ENABLE_MPI=ON' '-DPYTHON_EXECUTABLE:FILEPATH=/gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02
-23/stage_externals/install_gcc-12.2.0-skylake/python-3.10.8-rvn6l5/bin/python3.10' '-DNRN_ENABLE_MOD_COMPATIBILITY:BOOL=ON' '-DNRN_AVOID_ABSOLUTE_PATHS=ON' '-DCMAKE_C_FLAGS=-march=skylake -mtune=skylake'
```

Reply:
> but not for ~coreneuron (because in that case there is no gpu variant).

Didn't realise that variant now has `when` clause `when="@9: +coreneuron"`.

Remind me - we want to disable dynamic MPI only in the case of GPU, right? In that case something like:

```console
             if "+gpu" in self.spec:
                 args.append("-DNRN_ENABLE_MPI_DYNAMIC=OFF")
             else:
                 args.append("-DNRN_ENABLE_MPI_DYNAMIC=ON")
```
?

Reply:
I see two build failures in https://bbpgitlab.epfl.ch/hpc/cellular/nrn/-/pipelines/145606 with errors like:

```console
...
Failing in Thread:0
call to cuInit returned error 999: Unknown
```

Before setting debug builds and looking into details, @olupton or @iomaganaris - do you know if these could be relevant errors (from my NEURON change) or something irrelevant? Just in case you have seen this before.

Reply:
Unfortunately I don't have a clear idea regarding the source of the issue. However note that the `NEURON` CI is running with the `master` branch so maybe there is some issue introduced by some commit after yours

Reply:
Ok, no worries! 

Discussed with @olupton offline - from CI logs and Olli's observations, this issue has been mostly seen on Phase2 GPU nodes. Not sure about the reason.

We will look at this separately.